packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -380,32 +384,40 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
   ): Promise<FeeDataEstimate<KnownChainIds.SolanaMainnet>> {
     try {
       const { baseFee, fast, average, slow } = await this.providers.http.getPriorityFees()
+      const { sendMax, chainSpecific } = input
+      const { instructions } = chainSpecific
 
       const serializedTx = await this.buildEstimationSerializedTx(input)
-      const computeUnits = await this.providers.http.estimateFees({
+      const baseComputeUnits = await this.providers.http.estimateFees({
         estimateFeesBody: { serializedTx },
       })
 
+      const computeUnits = sendMax
+        ? bnOrZero(baseComputeUnits).times(SOLANA_COMPUTE_UNITS_BUFFER_MULTIPLIER).toFixed()
+        : baseComputeUnits
+
+      const instructionNumber = instructions?.length || SOLANA_MINIMUM_INSTRUCTIONS_NUMBER
```

Comment:
[nit]: naming

```suggestion
      const instructionCount = instructions?.length || SOLANA_MINIMUM_INSTRUCTIONS_NUMBER
```

Reply:
[preferably blocking]: Ensure the minimum value is never less than `SOLANA_MINIMUM_INSTRUCTIONS_NUMBER`

```suggestion
      const instructionNumber = Math.max(instructions?.length ?? 0, SOLANA_MINIMUM_INSTRUCTIONS_NUMBER)
```

Reply:
Ah it's actually a very nice catch, as providing the instruction could be prior the setComputeLimit and setComputeUnitPrice instructions that we can do later in the lifecycle, thanks for this @woodenfurniture 🐐 

Fixed in [275f332](https://github.com/shapeshift/web/pull/8327/commits/275f332ecfa01fe8a67b10465016c05190d26415)